packages/prices/utils/price.ts

```diff
@@ -115,26 +115,48 @@ export const getCurrency = (
 	};
 };
 
-const applyThousandSeparator = (
-	numberString: string,
-	thousandSeparator: string
+/**
+ * Get the integer value from the minor unit value.
+ *
+ * @param {number} priceInt The price in minor unit value, e.g. 100 for $1.00.
+ * @param {string} thousandSeparator The thousand separator.
+ * @param {number} minorUnit The number of decimals to display.
+ * @return {string} The extracted integer value.
+ */
+export const getIntegerValue = (
+	priceInt: number,
+	thousandSeparator: string,
+	minorUnit: number
 ): string => {
-	return numberString.replace( /\B(?=(\d{3})+(?!\d))/g, thousandSeparator );
+	return Math.floor( priceInt / 10 ** minorUnit )
+		.toString()
+		.replace( /\B(?=(\d{3})+(?!\d))/g, thousandSeparator );
 };
 
-const splitDecimal = (
-	numberString: string
-): {
-	beforeDecimal: string;
-	afterDecimal: string;
-} => {
-	const parts = numberString.split( '.' );
-	const beforeDecimal = parts[ 0 ];
-	const afterDecimal = parts[ 1 ] || '';
-	return {
-		beforeDecimal,
-		afterDecimal,
-	};
+/**
+ * Get the decimal value from the minor unit value.
+ *
+ * @param {number} priceInt The price in minor unit value, e.g. 100 for $1.00.
+ * @param {number} minorUnit The number of decimals to display.
+ * @return {string} The extracted decimal value.
+ */
+export const getDecimalValue = (
```

Comment:
These functions seem to be parsing the decimal value, **and** formatting it, is that correct? The `splitDecimal` I added seemed more easy to understand. Can we not do the formatting part (adding 0's etc) later using a separate function? cc @senadir